### PR TITLE
Add iterations runner for SOM VMs.

### PR DESCRIPTION
**Pull request by jacob-hughes**
Currently this does not support computing APERF / MPERF values.

### CODE REVIEW

File: iterations_runners/iterations_runner.som [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    krunGetWallclock: idx = primitive
+    krunGetCoreCyclesDouble: index core: core = primitive
+
+    run: args = (
+        self processArguments: args.
+        self runBenchmark.
+    )
+
+    usage = (
+      'Iterations runner for SOM VMs.' println.
```
Comment (ltratt) [inline iterations_runners/iterations_runner.som:18]: I *think* you can put an extra '\n' at the end of the strings if you want two blank lines? I haven't checked that in a while though...

File: iterations_runners/iterations_runner.som [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        instrument && (args length ~= 8) ifTrue: [ ^ self usage ].
+    )
+
+    loadBenchmarkClass: className = (
+        | sym cls |
+        sym := className asSymbol.
+        cls := system load: sym.
+        cls println.
+        cls ifNil: [
+            self error: 'Failed loading benchmark: ', className ].
```
Comment (ltratt) [inline iterations_runners/iterations_runner.som:51]: I think `,` is better as `+` (see https://github.com/SOM-st/SOM/pull/56).

File: iterations_runners/iterations_runner.som [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        bench := benchmarkClass new.
+        wallclockTimes := Array new: numIterations.
+        coreCycleCounts := Array new: numCores.
+        coreCycleCounts doIndexes: [:i | coreCycleCounts at: i put: (Array new: numIterations)].
+
+        i := 0.
+        [ i < numIterations ] whileTrue: [
+            | start end core |
+            debug > 0 ifTrue: [
+              ('[iterations_runner.som] iteration ' + i + '/' + numIterations) println.
```
Comment (ltratt) [inline iterations_runners/iterations_runner.som:69]: Indentation too small?

File: iterations_runners/iterations_runner.som [hunk 4]
```diff
@@ -0,0 +1,131 @@
+BenchmarkHarness = (
+
```
Comment (ltratt) [inline iterations_runners/iterations_runner.som:2]: Extraneous blank line?

**Comment by ltratt** [comment]
Just a few minor things.

File: iterations_runners/iterations_runner.som (hunk 1 above)
Comment (jacob-hughes) [reply iterations_runners/iterations_runner.som]: Ah cool, fixed: 35571d607cd13bb6c2f505424637621707105fa6

File: iterations_runners/iterations_runner.som (hunk 3 above)
Comment (jacob-hughes) [reply iterations_runners/iterations_runner.som]: Fixed: 8cf395c2c31fc03a32ca6a856e53aa5040b48a35

File: iterations_runners/iterations_runner.som (hunk 2 above)
Comment (jacob-hughes) [reply iterations_runners/iterations_runner.som]: Fixed: 5c0e33066356c2561d8281d72bb13f27bcbc829a

File: iterations_runners/iterations_runner.som (hunk 4 above)
Comment (jacob-hughes) [reply iterations_runners/iterations_runner.som]: Fixed: ff70d3309e4e2750ac2b2532274e22afce3a9cac

**Comment by jacob-hughes** [comment]
Thanks, I've fixed these.

File: iterations_runners/iterations_runner.som [hunk 5]
```diff
@@ -14,13 +14,10 @@ BenchmarkHarness = (
     )
 
     usage = (
-      'Iterations runner for SOM VMs.' println.
-      '' println.
-      'Executes a benchmark many times within a single process.' println.
-      '' println.
+      'Iterations runner for SOM VMs. \n' println.
```
Comment (ltratt) [inline iterations_runners/iterations_runner.som:17]: Are the trailing spaces at the end of lines intentional (after the `.` and before the `\n`)?

Comment (jacob-hughes) [reply iterations_runners/iterations_runner.som]: No, they're the result of me not thinking. Try this: e4fbf31a45a8fcc68c4d35fe63ade2541e0062ea

**Comment by ltratt** [comment]
Please squash.

**Comment by jacob-hughes** [comment]
Squashed

**Comment by ltratt** [comment]
Ah, this isn't a bors repo :/

**Review by ltratt — APPROVED** [review]

**Pull request merged**